### PR TITLE
Feature/update position query

### PR DESCRIPTION
**Pull request by gamedolphin**

### CODE REVIEW

**Review by kingstone426 — APPROVED** [review]
Looks good!

File: src/server/send_query.go [hunk 1]
```diff
@@ -70,29 +71,34 @@ func newSendQuery(completeSourceWorld *ecs.World, userEntity ecs.Entity, targetW
...
-
-	// The transfer query must be added after the position query, or it will begin transferring entities too soon
-	transferQuery := newTransferQuery(b.participant, b.transferManager, b.log)
-	b.query.Queries = append(b.query.Queries, transferQuery)
+	if b.positionQueryIndex != -1 {
+		positionQuery := b.query.Queries[b.positionQueryIndex].(*ecsquery.TestDistanceQuery)
+		positionQuery.UpdateQuery(position, radiusFixedPoint)
+		b.query.Queries[b.positionQueryIndex] = positionQuery
+
+		b.log.Debug("updating position query")
```
Comment (rpadrela) [inline src/server/send_query.go]: I would extract this to UpdatePositionQuery().

File: src/server/send_query.go [hunk 2]
```diff
@@ -70,29 +71,34 @@ func newSendQuery(completeSourceWorld *ecs.World, userEntity ecs.Entity, targetW
...
+		worldPositionQuery := ecsquery.NewTestDistanceQuery(b.completeSourceWorld.EntityManager(),
+			archetypeForUnderlyingQuery, standardTransformTypeID,
+			0, position, radiusFixedPoint, b.log)
+		b.query.Queries = append(b.query.Queries, worldPositionQuery)
+		b.positionQueryIndex = queryIndex
+		// The transfer query must be added after the position query, or it will begin transferring entities too soon
+		transferQuery := newTransferQuery(b.participant, b.transferManager, b.log)
+		b.query.Queries = append(b.query.Queries, transferQuery)
+
+		b.log.Debug("setting up position query")
```
Comment (rpadrela) [inline src/server/send_query.go]: I would extract this to CreatePositionQuery().

**Review by rpadrela — APPROVED** [review]
LGTM

**Pull request merged**